### ISSUE TITLE
Slash for invalid or fake Ethereum events

### ISSUE THREAD

**Issue by james-chf** [anoma/namada#180]
(relevant version of spec: https://github.com/anoma/namada/blob/c6bc20d35497d0244c201c698e79471aca2027c0/documentation/spec/src/interoperability/ethereum-bridge.md)

Validators may construct invalid Ethereum events and vote to include them in Namada. For example, they could construct a fake batch of ERC20 transfers. We need to ensure at least the following:
- validators who vote for invalid Ethereum events SHOULD be slashed before they can unbond
- invalid/fake Ethereum events MUST NOT be acted upon (in the case where 2/3 of total voting power is not colluding on such)

Currently, we don't include anything in the Ethereum events put in vote extensions that would allow them to be checked as valid or invalid with regards to the Ethereum chain - e.g. the Ethereum block hash where the event was emitted, or an Ethereum proof of inclusion.

We should include something like this, then a block proposer could include signed Ethereum events they consider invalid alongside ones they consider valid. (Honest) validators can check both of these sets during `ProcessProposal`. Where an event is agreed as invalid, the validator(s) who voted for it should immediately be marked for slashing. Where a block proposer marks a valid event as invalid, the block proposer should be slashed. Invalid events should then be included on chain as proof of the slashable behaviour that occurred.  In any case, it should not be possible for invalid events to be stored in `/eth_msgs` and then voted on to be acted upon. There may be liveness/performance implications to doing it this way though as it means a block proposer and validators must carry out validation beyond the syntactic on Ethereum events in vote extensions.

This issue covers
- [ ] updating the spec
- [ ] making updates to the code in `eth-bridge-integration`, or otherwise replacing this issue with smaller broken down issues to do so

**Comment by cwgoes** [comment]
@batconjurer do you know the status of this?

**Comment by batconjurer** [comment]
@cwgoes Actually, I would consider this no longer relevant. When we still used an Ethereum light client in the design, we could check for fake events. Without that, for an attacker with even a modicum of sophistication, we can no longer tell. We just rely on getting a quorum of backing signatures. Thus we can't really enforce slashing. 

We discussed this together back when we changed the design re: security concerns, dos vectors etc. I assume you are still of the same mind?
